### ISSUE TITLE
Open modal is shifting fixed navbar to the right

### ISSUE THREAD

**Issue by mhawk0** [twbs/bootstrap#14040]
re: https://github.com/twbs/bootstrap/issues/9855
in 3.2.0, when body has a scrollbar, it's content now stays in place when modal is opened (padding is added to the body).
But fixed top navbar, having position: fixed, left: 0 and right: 0, now jumps to the right when modal is opened, as it ignores padding-right on the body.

**Comment by restran** [comment]
@parminderkaur

This fixes for me in version 3.2.0

.modal-open[style="padding-right: 17px;"] .navbar-fixed-top,
.modal-open[style="padding-right: 17px;"] .navbar-fixed-bottom {
    padding-right: 17px;
}

**Comment by M1chae1** [comment]
@parminderkaur Thanks! it works

**Comment by Hint-ru** [comment]
@ramon18 Thx. But we should check this.bodyIsOverflowing too.
`if (this.bodyIsOverflowing && this.scrollbarWidth) $('.navbar-fixed-top').css('padding-right', this.scrollbarWidth);`

**Comment by JiNexus** [comment]
I'm still experiencing this kind of bug in Bootstrap v3.3.4 I hope they will fix this issue. I tried putting padding-right: 17px on the fixed navbar to temporarily fix it but I found out that it only works on Chrome and Firefox using windows OS but when I switch to Ubuntu OS the value of padding-right in the body when modal is opened is changed to 15px, so that means that I have to change the value that I set in fixed navbar from 17px to 15px just to fix it on Chrome and Firefox in Ubuntu platform.

**Comment by scooterlord** [comment]
Hello. 

After putting up with this for a couple of years and using hacks, shivs and useless javascripts, here I come with a pure css solution to the problem. Simple as hell. Cross-browser.

We create a class (eg no-jump), and apply it to all direct descendands of body and fixed elements.

Then simply:

.modal-open .no-jump {
  overflow-y:scroll;
}

Since the body overflow is hidden and usually fixed elements are either at top or bottom and noone would scroll there, noone will ever notice it in any browser.

It's cross-browser, which means it doesn't need to detect scrollbar width which is different among different browsers/OS'es, OR detect touchscreens that have no scrollbars at all. Overflow-y will create the appropriate width of scrollbar according to each occasion.

I am surprised no-one came up with this solution so far (including myself).

**Comment by ramon18** [comment]
@Hint-ru Ya the user may resize the window between modals, that's a nice fix, thanks :+1:

**Comment by JiNexus** [comment]
Thanks @ramon18 and @Hint-ru you did great fixing with this kind of issue.

**Comment by nozpheratu** [comment]
@Hint-ru @JiNexus @ramon18 Unfortunately that won't work on different browsers / operating systems as scrollbars won't always have a pixel width of 17px. For instance, it's only 15 on Ubuntu Firefox.

**Comment by JiNexus** [comment]
Actually I tried it on Windows and Ubuntu, both Firefox and Chrome. They work to me perfectly fine. I don't know with Mac or any other browser such as Opera.

**Comment by nozpheratu** [comment]
@JiNexus Just saying, if I was your user and I went to your website right now using those snippets your website would still have the problem for me. It might work on some browsers/OSs, but it's no silver bullet.

**Comment by JiNexus** [comment]
@nozpheratu if you saw my previous post I already told that kind of problem that in different OS's the pixel is not always 17px. By the way have you seen the code that we are using? If you notice we are not defining a value to the padding-right, we let Javascript define the value by using "scrollbarWidth" so that it will be flexible to other OS's or browsers. Please review the code before jumping out to conclusions.

**Comment by nozpheratu** [comment]
@JiNexus I actually cc'd the wrong person, I meant to direct that original post to @restran.

**Comment by JiNexus** [comment]
@nozpheratu Seriously? you accidentally cc'd 3 wrong person from your original post? Anyway it doesn't matter, this case is close I hope bootstrap will add this fix to their next update.

**Comment by mrsheepsheep** [comment]
I fixed it using

.modal-open[style] {
   padding-right: 0px !important;
   overflow-y:scroll;
}

**Comment by JiNexus** [comment]
That code is so static. At first glance I know there's a bug on it like for
example if the modal would have a heap of content. The modal wouldn't be
scrollable anymore.

**Comment by nhouse** [comment]
This issue is a continuation of https://github.com/twbs/bootstrap/issues/9855 which was opened about a year and a half ago. Considering the number of websites using Bootstrap and affected by this probem, I'm astounded it hasn't been fixed yet.

**Comment by JiNexus** [comment]
I don't recommend those css hacks that puts overflow: y-scroll, lets say for example your modal has a heap of content on it that means you wont be able to scroll down your modal anymore. 
For me the right fixed is this, you allow Javascript to determine its scrollbarWidth and put it as your padding-right and as the modal closes you also reset the padding-right by removing the css: (Credit to @ramon18  )
Note that I didn't include fixed .navbar-fixed-bottom since I'm not using it. You can include it if you are using it.

/* File: fixed.js */
/* Fix shifting fixed navbar to the right */
$(document).ready(function(){
	$(window).load(function(){

		var oldSSB = $.fn.modal.Constructor.prototype.setScrollbar;
		$.fn.modal.Constructor.prototype.setScrollbar = function () 
		{
			oldSSB.apply(this);
			if(this.bodyIsOverflowing && this.scrollbarWidth) 
			{
				$('.navbar-fixed-top').css('padding-right', this.scrollbarWidth);
			}		
		}
		
		var oldRSB = $.fn.modal.Constructor.prototype.resetScrollbar;
		$.fn.modal.Constructor.prototype.resetScrollbar = function () 
		{
			oldRSB.apply(this);
			$('.navbar-fixed-top, .navbar-fixed-bottom').css('padding-right', '');
		}

	});
});

**Comment by emzero** [comment]
@scooterlord Can you provide a full example? I've added your CSS class and applied it to my nav element which is the only fixed element but it's still shifting when the modal is opened...

**Comment by nhouse** [comment]
@JiNexus's code worked great for me.

**Comment by sarahmurray** [comment]
A similar issue occurs when immediately opening a modal from another modal (using "show" and "hide" methods) – the padding and .modal-open class are removed from the body but not re-added, until the second modal is dismissed, when the padding is then added to the body. And if you go through the loop a few times it keeps adding more and more padding to the body.

Or is opening a modal from another modal not supported?

**Comment by ramon18** [comment]
Overlapping Modals is not supported
See http://getbootstrap.com/javascript/#modals

**Comment by JiNexus** [comment]
@sarahmurray I suggest you try to use the event handling in bootstrap http://getbootstrap.com/javascript/#modals-events, like for example
hidden.bs.modal - This event is fired when the modal has finished being hidden from the user (will wait for CSS transitions to complete).

In my sample code below, My #dangerModal won't show until the #loadingModal is hidden. Please always use $(document).ready(function(){ .. }); and $(window).load(function(){ .. }); this is to ensure that all js are loaded.

$(document).ready(function(){
	$(window).load(function(){
		$(function(){
			$(\'#loadingModal\').on(\'hidden.bs.modal\', function (e) {
				$(\'#dangerModal\').modal({show: true,
									   keyboard: false,
									   backdrop: \'static\'});
			});
		});
	});
});

**Comment by eLement87** [comment]
I had the same issues and no solutions above worked for me.

But this piece of code worked for me to fix the 15px bar on the right side :

```
body {
    padding-right: 0 !important;
}
```

But the Main-Nav is still jiggeling, when the Modal is comming up. 
And then...

```
#main-nav {
    padding-right: 0 !important;
}
```
And it's gone :)

Bootstrap 3.3.4

**Comment by unholyknight** [comment]
@eLement87 Those definitions look like they work without any unintended consequences as far as I've noticed. Thanks!

**Comment by blackfyre** [comment]
:+1: For @eLement87's solution, and @unholyknight's observations

**Comment by silvenon** [comment]
@JiNexus' solution was adding padding to avoid shifting to the right. I don't see how the solution above could possibly work as it brings everything, including Bootstrap's own patches, back to square zero, negating that the issue even exists. Am I missing something? Are you sure you were testing in environments where scrollbars take up actual space?

**Comment by ryoga7482** [comment]
<pre><code>body.modal-open {
overflow-y: scroll;
padding-right: 0 !important;
}</code></pre>

Worked in my situation - combination of this and <a href="https://github.com/twbs/bootstrap/issues/9855" class="issue-link" title="Open modal is shifting body content to the left">#9855</a>

**Comment by fxrising** [comment]
Thanks ryoga7482, that was the only css solution that worked for me :)

**Comment by nairys** [comment]
@ryoga7482 worked for me. thanks!

**Comment by JosKrause** [comment]
Also would like to add that @ryoga7482 solution worked absolutely flawless so far in all of my pages dealing with modals, also when the content is a centered container; applied to BS 3.3.0.

**Comment by schack86** [comment]
Hi guys. I'm new on this forum, but I'm experiencing the same problem with bootstrap 3.3.5. Implementing this solution:
body.modal-open {
    overflow-y: scroll;
    padding-right: 0 !important;
}
fixes the issue of added padding to body perfectly. But my modal isn't horizontally centered anymore. It's moved to the left by amount of the width of the scrollbar (in my case: padding-right: 17px;)
Does any of you brainiacs know how to fix it?

**Comment by JiNexus** [comment]
@schack86 try the our code above. I assure it work perfectly fine in different browser and OS.

**Comment by schack86** [comment]
@JiNexus I've tried your code in win7 chrome and firefox, and so far it works like a charm. I have yet to try it on ios.

**Comment by JiNexus** [comment]
@schack86, I've been there in your situation. We are glad that it fixes your problem. I don't want to argue with the other fixes above since only those who does a split testing with different browser with different OS's will truly understand which one is the right fix for them.

**Comment by schack86** [comment]
@JiNexus Am I missing something. The fixed top nav still works perfectly. But if the modal is shorter than the viewport and scrollbar isn't needed, then the padding on different elements still occur. Do I have to include these elements somehow in the script? Please help.

I simply don't understand why this issue haven't been solved by the bootstrap team yet. So frustrating!

**Comment by JiNexus** [comment]
@schack86, Hey, I reviewed my code and my modal, I also tried to lessen the contents of my modal to make it shorter in the screen and see if I will experience the same issue that you are talking about but so far it is working fine on my end. Can you try to clean the css fixed that you have made before you use our code to fix the problem with modal? It might be the culprit. For clarification I'm currently using Bootstrap v3.3.4. Cheers!

**Comment by schack86** [comment]
@JiNexus Back to basics. I've tried making a simple jsfiddle snippet with bootstrap 3.3.5 and jquery 2.1.3 and implementing your code. The only css besides bootstrap is colouring the divs. Both modals isn't centered 100% and the navbar shifts place on open and close of modal.

https://jsfiddle.net/d4bg8uzj/2/embedded/result/

Could you please tell me what I am doing wrong?

**Comment by twbs-lmvtfy** [comment]
Hi @schack86!

You appear to have posted a live example (https://fiddle.jshell.net/d4bg8uzj/2/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 141, column 9 thru line 141, column 50: Duplicate ID ```` myModalLabel ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by schack86** [comment]
@JiNexus I tried stripping the code down to the bare minimum and I noticed now that the problem occurs at the breakpoint around 992px. 1920 and down to 993 works perfectly, but when it hits 992 the padding is added to the body with a short modal. The long scrollbar of the long modal hides the problem. Can you confirm this bug?

You can check it out here:
http://modal.martinschack.dk/

I hope you understand my problem :)

**Comment by JiNexus** [comment]
@schack86 I did reviewed your sample site that you've provided, thank you for that it makes things much easier. Back to the main issue, I've found out that it's not our code that is causing the problem, actually it solve the issue of "modal shifting fixed navbar to the right" at the very first place. The main culprit of your problem is the class introbox that you have in your div which has a css of (background: #0f0 none repeat scroll 0 0;). Please try to remove it and see it to yourself. Cheers!

**Comment by schack86** [comment]
@JiNexus I simply don't understand it. The only thing that I, myself, is adding to the css is the background color. The "no-repeat scroll 0 0;"-part must be added by bootstrap or some other part. When I use chrome devtools I don't see the code you're talking about. But nevertheless, I don't know how to fix it. This is driving me nuts! Could you please take a screenshot and paste it here?

**Comment by JiNexus** [comment]
@schack86 I think the problem is the way you handle your layouts. This is just a sample layout that I extracted in my template, feel free to try it out. Cheers!
```
<body style="background-color: green;">
	<header>
		<nav class="navbar navbar-inverse navbar-fixed-top">
			<div class="container-fluid">
				<div class="navbar-header">
					<button class="navbar-toggle collapsed" aria-expanded="false" data-target="#bs-example-navbar-collapse-6" data-toggle="collapse" type="button">
						<span class="sr-only">Toggle navigation</span>
						<span class="icon-bar"></span>
						<span class="icon-bar"></span>
						<span class="icon-bar"></span>
					</button>
					<a class="navbar-brand" href="#">Brand</a>
				</div>
				<div id="bs-example-navbar-collapse-6" class="collapse navbar-collapse">
					<ul class="nav navbar-nav">
						<li class="active">
							<a href="#">Home</a>
						</li>
						<li>
							<a href="#">Link</a>
						</li>
						<li>
							<a href="#">Link</a>
						</li>
					</ul>
				</div>
			</div>
		</nav>
	</header>

	<div class="container">
		<div class="row">
			Your contents here!!!!!!!!!!!!!!!!! (START)
			<button type="button" class="btn btn-primary btn-lg" data-toggle="modal" data-target="#myModal">
				Launch demo modal
			</button>

			<!-- Modal -->
			<div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
				<div class="modal-dialog" role="document">
					<div class="modal-content">
						<div class="modal-header">
							<button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
							<h4 class="modal-title" id="myModalLabel">Modal title</h4>
						</div>
						<div class="modal-body">
						...
						</div>
						<div class="modal-footer">
							<button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
							<button type="button" class="btn btn-primary">Save changes</button>
						</div>
					</div>
				</div>
			</div>
			Your contents here!!!!!!!!!!!!!!!!! (END)
		</div>
	</div>

	<footer>
		<div class="footer" style="background-color: blue;">
			<div class="container">
				<p class="text-muted">Place sticky footer content <a href="http://getbootstrap.com/examples/sticky-footer/">here</a>. </p>
			</div>
		</div>
	</footer>
</body>
```

**Comment by adrianparr** [comment]
This worked for me ...

	body.modal-open {
		overflow-y: auto !important;
		padding-right: 0 !important;
	}

**Comment by schack86** [comment]
@JiNexus ... I've tried the code you've send, including your fixed.js. It works perfectly if the modal is light on content. BUT when the modal is filled with a lot of content and its height is greater than the viewport, the padding-left is added, and the modal is off again... Haven't included anything but your code and the bootstrap base. It's especially noticeable at breakpoint 768px. Please confirm that you experience this as well.
![untitled-3](https://cloud.githubusercontent.com/assets/13488437/9231223/5751f7b0-4127-11e5-8e07-460ade543bca.jpg)

**Comment by Praveen2910** [comment]
![github](https://cloud.githubusercontent.com/assets/14801815/10041542/7c9b5c2c-6202-11e5-9d27-13263dc16140.png)

Simply make the required changes in the bootstrap.css file to fix the issue of the modal shifting the entire site to left or adding a padding to the site.

**Comment by erickacevedor** [comment]
@ryoga7482 Solution works for me too.

**Comment by safderjaffri** [comment]
Hi everyone, here's my solution its working for me
1. open > bootstrap.min.css OR bootstrap.css
2. find class 
.modal-scrollbar-measure{
position:absolute;
top:-9999px;
width:50px;
height:50px;
/*overflow:scroll this is bootstrap default value change it with*/  overflow:hidden;
}
3. find another class.
.modal-open {
    /*overflow:hidden this is bootstrap default value change it with*/ overflow:visible;
}
for mobile device:
.modal-open {
     overflow:hidden;
}

thanks

**Comment by tallpaullewis** [comment]
I 'fixed' mine with a combination of the above:
```
body.modal-open{
    width:100%!important;
    padding-right:0!important;
    overflow-y:scroll!important;
    position:fixed!important;
}

**Comment by vijayviji** [comment]
The given solution doesn't seem to work on 'edge' browser

**Comment by sebwebdesigner** [comment]
@sanjay900 Nice solution, and the only one that worked for me. 
The navbar still shifted when closing the modal so I changed hide.bs.modal to hidden.bs.modal and it did the trick.

**Comment by gabrieldrs** [comment]
My solution is as follows:
```
body[style*="padding-right: 17px"] .navbar-fixed-top {
  padding-right:17px;
}
```

The problem is that bootstrap adds a "padding-right" style to the body element when a vertical scroll bar is present, but since the navbar is fixed it won't respect that, so the style above applies only to the fixed navbar that is children of a body that contains the "padding-right" property in its style attribute.

I had to do that because the solutions posted before this one adds or remove this padding to the page regardless if it has a scroll bar or not, and I had really short pages that was making the design break.

OBS.: If your fixed navbar is not a "navbar-fixed-top", just replace it with the proper class of your navbar.

Hope this helps someone out there :)

**Comment by redeventmedia** [comment]
@tallpaullewis yup that worked for me too! Thanks.

**Comment by tehXor** [comment]
Beaware of the @ryoga7482 "solution" and others that hard code CSS properties. In the case of the solution with a fixed `overflow-y: scroll;` on `body` you will get the effect that when your modal has enough content that it also overflows the screen there will be two scrollbars. And this is usability wise very poor. If you call that a solution well than so be it. But I guess for most people it is not and they actually may not know that they run into this kind of problem when they test with small modals on big screens. On mobile devices most your modals will overflow even if they don't have much content...

**Comment by JiNexus** [comment]
@tehXor I agree, guys please consider the solution that I made, just scroll up. I've implemented it to my app and so far it fixes my issue dynamically.

**Comment by Eagle3386** [comment]
@JiNexus what about @eLement87's suggestion in his comment? Isn't that valid, too?
(Don't know how to reference a comment in Markdown, so here's the full URL: https://github.com/twbs/bootstrap/issues/14040#issuecomment-96361083)

**Comment by JiNexus** [comment]
@Eagle3386 His solution is not dynamic you will probably understand when you encounter same problem with `<a href="https://github.com/twbs/bootstrap/issues/14040#issuecomment-124684418">#14040</a>`. I don't recommend a css hack on the modal issue since it's basically a javascript driven. A static css hack might work on a certain browser but not to all. If you follow the thread and reread the comments you will understand how we arrive with that solution.

**Comment by Eagle3386** [comment]
@JiNexus I do get your point and it's an absolutely valid one. I'll stick with the jQuery one then, too.
Thanks for the clarification and help, though! :)

Hopefully, this gets added to Bootstrap soon.

**Comment by stosh15x** [comment]
This worked for me:

`$('#myModal').bind('hidden.bs.modal', function () {
    $("body").css("padding-right", "0px");
    $('#myModal').modal('handleUpdate');
  });
  $('#myModal').bind('show.bs.modal', function () {
    $("body").css("padding-right", "-15px");
    $('#myModal').modal('handleUpdate');
  });`

**Comment by saibalroy** [comment]
my bootstrap modal form jump on safari ipad.how to fix it?

**Comment by darrelllance** [comment]
This worked for me:

Removed this in boostrap.min.js:
`&&this.$body.css("padding-right",a+this.scrollbarWidth)`

And added this to my CSS:
`body.modal-open {
    padding-right: 0px !important;
}
html {
    overflow-y: auto !important;
}`

**Comment by saibalroy** [comment]
where i add this css?

**Comment by darrelllance** [comment]
Your custom css.

**Comment by saibalroy** [comment]
it's not working.

**Comment by saibalroy** [comment]
my form still jumping.

**Comment by Eagle3386** [comment]
Are you guys willing to scroll just 5 posts above? @JiNexus was kind enough to repeat the working (!) solution there.

@JiNexus thx for the detailed explanation. I'll go with your solution. ;) :+1:

**Comment by leoj3n** [comment]
+1

**Comment by kapilindore** [comment]
/*This worked for me ---Bootstrap v3.3.6---
---after bootstrap.css---*/
body.modal-open {
    overflow-y: auto !important;
    padding-right: 0 !important;
}

.modal-scrollbar-measure {
  
  overflow: hidden;
}

**Comment by vikarz** [comment]
have the same problem on bootstrap 3.3.6, layout shrink to left after showing second modal, please help. i'm developing web for school.

**Comment by xuefengwang** [comment]
@JiNexus Thank you for the solution. Using already calculated scroll bar width looks a good fix.

**Comment by JiNexus** [comment]
@xuefengwang No problem, this fixed is also credited to @ramon18 & @Hint-ru 
Cheers!

**Comment by vickyraj1521** [comment]
This works fine

body.modal-open{
    width:100%!important;
    padding-right:0!important;
    overflow-y:scroll!important;
    position:fixed!important;
}

**Comment by mizurnix** [comment]
@vickyraj1521 It does, unless you have a long modal. If modal content needs scrolling, then you have two scrollbars and it's ugly

**Comment by vickyraj1521** [comment]
You may try after setting a min height for the page content. I hope it will solve the problem.

**Comment by erickacevedor** [comment]
Hey @JiNexus, i'm getting this error in Chrome console when i'm using your solution: 
`Uncaught TypeError: Cannot read property 'Constructor' of undefined` 

And the code is exactly the same you provide. This is how appears in the Chrome inspector: 
![screen shot 2016-06-26 at 9 31 29 am](https://cloud.githubusercontent.com/assets/2944957/16362932/d41d03d0-3b80-11e6-816e-3f5130f7a98d.png)

I'm using BS 3.3.6 in Google Chrome w/ OSX Yosemite.

**Comment by JiNexus** [comment]
@erickacevedor Sorry, I've been in the back end for a couple of months now and never touch some front end works for awhile so I don't know what's the latest updates of Bootstrap now. But I guess the issue is you load it before bootstrap, therefore it assumes bootstrap is already loaded. Append it in the right order: jQuery -> Bootstrap -> Your Fix

I hope my assumptions helps you. :D

**Comment by erickacevedor** [comment]
Hey @JiNexus, indeed helped, now is fixed. Thank you a lot! 🎉

**Comment by dejwne** [comment]
@mizurnix I have the same issue, did you figure out how to hide a second scroll bar? Thank you

**Comment by mizurnix** [comment]
@dejwne I went with the other [solution ](https://github.com/twbs/bootstrap/issues/14040#issuecomment-89720484) provided above by @JiNexus. It seems to cover most cases (to this point, all cases I could come up with) and work well cross-browser.

**Comment by dejwne** [comment]
Could you help me @mizurnix od @JiNexus please? I'm using exactly same code from @JiNexus added to my file fixedModal.js and loading it after jQuery and Boostrap, but nothing change. The page is still moving to the right after modal is shown.

Boostrap v4.0.0-alpha.2
Chrome 51.0.2704.106

**Comment by danieldafoe** [comment]
Is this still an issue for anyone? I've tested in Safari 10 and Chrome 52 on macOS and can't reproduce when opening a modal when there's a navbar with `.navbar-fixed-top` on the page.

**Comment by inge1980** [comment]
Thanks @JiNexus! Beautifully done :+1: 

>     $(document).ready(function(){
        $(window).load(function(){
            var oldSSB = $.fn.modal.Constructor.prototype.setScrollbar;
            $.fn.modal.Constructor.prototype.setScrollbar = function () {
                oldSSB.apply(this);
                if(this.bodyIsOverflowing && this.scrollbarWidth) {
                    $('.navbar-fixed-top, .navbar-fixed-bottom').css('padding-right', this.scrollbarWidth);
                }       
            }
            var oldRSB = $.fn.modal.Constructor.prototype.resetScrollbar;
            $.fn.modal.Constructor.prototype.resetScrollbar = function () {
                oldRSB.apply(this);
                $('.navbar-fixed-top, .navbar-fixed-bottom').css('padding-right', '');
            }
        });
    });

**Comment by xbroak** [comment]
Still an issue for me yes!

**Comment by rmcsharry** [comment]
I still see this problem using Bootstrap v3.3.7

MacOSx El Capitan in
Chrome 52.0.2743.116
Firefox 48.01
Safara 9.1.1

[in html or in body selector] {
overflow-y: scroll; 
}

That will trigger it with standard bootstrap setup when the modal opens.

MacOS has scrollbars set to 'always on' in system prefs.

I tried the js fix from @JiNexus and sadly did not work on this platform. But he is correct that the various CSS hacks will only work in certain platform/browser/application combinations.

**Comment by smQ93** [comment]
I have faced to this problem too and I found a css workaround:

body.modal-open {
  padding-right: 0 !important;
}

Tested on Windows 7: Chrome 52/ Mozilla 48/ IE9 and works fine.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team

**Comment by Goloveichuk** [comment]
My jQuery solution:

```
var nb = $('nav.navbar-fixed-top');
$('.modal')
    .on('show.bs.modal', function () {
        nb.width(nb.width());
    })
    .on('hidden.bs.modal', function () {
        nb.width(nb.width('auto'));
    });
```

**Comment by maslick** [comment]
Same thing in Chrome (Ubuntu). However, Chrome and Safari on MacOS didn't have this 15px shift.
Remedy that worked for me:
```
body {
  padding-right: 0 !important;
}
```

**Comment by Eagle3386** [comment]
The _only_ solution that works cross-plattform is the one provided by JiNexus [above](https://github.com/twbs/bootstrap/issues/14040#issuecomment-89720484). So, deal with it instead of repeating the same padding-hacks as others did before you. All subscribed users will thank you in advance!

**Comment by enekogb** [comment]
The fix provided by @ramon18 [here](https://github.com/twbs/bootstrap/issues/14040#issuecomment-52120453) and @JiNexus [here](https://github.com/twbs/bootstrap/issues/14040#issuecomment-89720484 works like a charm but ....

If you have a close-one-modal-and-open-another-modal action, be sure to wait until the first modal closes before opening the second one, or it won't work.

```
// BAD
modal1.modal('hide');
modal2.modal('show');

// BETTER
modal1.one('hidden.bs.modal', function() {
  modal2.modal('show');
}).modal('hide');
```

**Comment by bke-daniel** [comment]
Sadly, I'm still experiencing this issue in BS4, Alpha 6

**Comment by oddlyfunctional** [comment]
+1

**Comment by Eagle3386** [comment]
@bke-daniel this issue is for the "old" BS3 branch, hence your comment is a) in the wrong place, b) doesn't add useful information to the issue (the working fix is explained and linked above within the posts of @enekogb and myself) and c) almost only spam regarding the huge amount of subscribers being notified for no good reason.

@oddlyfunctional There's a +1 feature provided by GitHub, so spare us subscribers with such posts. Especially since my post 4 posts before explained everything neccessary.

**Comment by vishalmule17** [comment]
@adrianparr  your CSS work for me ..thanks buddy

**Comment by DeepakBisht94** [comment]
.modal-open .navbar{margin-right: 17px; }

**Comment by emmzone** [comment]
After many hours of reading and testing, this worked for me:

var scr_size = window.innerWidth;
var scr_avail = $('body').innerWidth();
var pad_right = scr_size - scr_avail;

$('.modal')
    .on('show.bs.modal', function () {
		$('.YourFixedNavClass').css('padding-right', pad_right);
    })
    .on('hidden.bs.modal', function () {
        $('.YourFixedNavClass').css('padding-right', '');
    });

I hope this helps someone else.

Cheers.

**Comment by Eagle3386** [comment]
@emmzone, no, this will not fix the bug cross-platform - as was already stated several times before.
I find your lack of respect to the followers of this issue disgusting. Please remove your comment so others won't be confused or even mislead as it's simply false.

The one and only solution to this bug still remains to be the code [posted above](https://github.com/twbs/bootstrap/issues/14040#issuecomment-89720484) by @JiNexus - deal with it!

To @mhawk0, @cvrebert or anybody other person with appropriate permissions: can you lock this issue so that further inoperable, non-cross-platform and useless comments won't get added to it?
By doing so, please either delete this comment afterwards, too or drop me a line so that I remove it myself.

To every other notified person: sorry for the noise, just trying to stop this mess.
